src/tool_parsecfg.c

```diff
@@ -335,39 +343,17 @@ static const char *unslashquote(const char *line, char *param)
 
 /*
  * Reads a line from the given file, ensuring is NUL terminated.
- * The pointer must be freed by the caller.
- * NULL is returned on an out of memory condition.
+ * FALSE is returned on an out of memory condition.
  */
-static char *my_get_line(FILE *fp)
+static bool my_get_line(FILE *fp, struct curlx_dynbuf *db)
 {
   char buf[4096];
-  char *nl = NULL;
-  char *line = NULL;
-
   do {
-    if(NULL == fgets(buf, sizeof(buf), fp))
-      break;
-    if(!line) {
-      line = strdup(buf);
-      if(!line)
-        return NULL;
-    }
-    else {
-      char *ptr;
-      size_t linelen = strlen(line);
-      ptr = realloc(line, linelen + strlen(buf) + 1);
-      if(!ptr) {
-        Curl_safefree(line);
-        return NULL;
-      }
-      line = ptr;
-      strcpy(&line[linelen], buf);
-    }
-    nl = strchr(line, '\n');
-  } while(!nl);
+    if(!fgets(buf, sizeof(buf), fp) ||
+       curlx_dyn_add(db, buf))
+      /* failure */
+      return FALSE;
```

Comment:
fgets will return 0 if there are no more characters to be read, so that should not be treated as an error. You could break on feof like I suggested yesterday, which would then cause the function to return true indicating a line has been written, however it occurs to me it's possible fgets has only been called that single time therefore it's possible nothing was added to the buffer.

Reply:
Yes a zero length file will still return success with nothing written to it, but I think that's fine and expected.

Reply:
To be clear what I meant is a fgets a single time in the call to my_get_line but fgets could have been called before and is going to return true after the last line if you break. Say you have a file with one character
my_get_line(fp, db) ---> returns true and two bytes have been written, the byte from the file and 0
my_get_line(fp, db) ---> returns true (due to break) and nothing has been written

Reply:
I don't understand. Are you saying the current incarnation of the function has a problem or not?